### ISSUE TITLE
feature: Add x-axis interpolator

### ISSUE THREAD

**Issue by paucablop** [paucablop/chemotools#261]
# Feature Request: XAxisInterpolator Transformer

## Is your feature request related to a problem? Please describe.

When working with spectral data, sensor measurements, or other signals where samples are acquired on different input grids (e.g., wavelength, frequencies, time points), there's a need to resample all samples onto a common output grid before passing them to downstream estimators in a scikit-learn pipeline.

Currently, this preprocessing step must be applied manually **before** feeding data to a pipeline, which breaks the standard scikit-learn workflow. Users cannot:
- Use `GridSearchCV` or `cross_validate` with custom grid parameters
- Apply the resampling within a `Pipeline` where metadata can be properly routed
- Leverage scikit-learn's metadata routing system to pass sample-specific input grids through estimators

## Describe the solution you'd like

A general **interpolation transformer** that:

1. **Resamples input features** from sample-specific (or shared) input grids to a fixed common output grid
2. **Supports scikit-learn metadata routing** so sample-specific grids can flow through pipelines and grid search
3. **Integrates with scikit-learn's standard estimator interface** (BaseEstimator, TransformerMixin)
4. **Handles edge cases** like different input grid sizes per sample, values outside the interpolation domain
5. **Supports multiple interpolation algorithms** (linear, spline, etc.) via a configurable interpolation method

This would enable typical workflows like:

```python
from sklearn.pipeline import make_pipeline
from sklearn.preprocessing import StandardScaler
from sklearn.cross_decomposition import PLSRegression

# Enable metadata routing
sklearn.set_config(enable_metadata_routing=True)

# Build pipeline with interpolator
pipe = make_pipeline(
    XAxisInterpolator(output_grid=common_x_axis)
    .set_fit_request(input_grid=True)
    .set_transform_request(input_grid=True),
    StandardScaler(),
    PLSRegression(n_components=10)
)

# Use in GridSearchCV with sample-specific grids
grid = GridSearchCV(
    pipe, 
    param_grid={"xaxisinterpolator__n_jobs": [1, -1]}
)
grid.fit(X_train, y_train, xaxisinterpolator__input_grid=x_axis_train)
```

## Describe alternatives you've considered

**Manual preprocessing outside the pipeline**: Apply interpolation before creating the pipeline. This works but prevents use with `GridSearchCV`, `cross_validate`, and breaks reproducibility.

## Additional context

- This transformer would be particularly valuable for scientific data (spectroscopy, chromatography, sensor data)
- Metadata routing (available in sklearn >= 1.3) makes this pattern feasible and robust
- The transformer should be flexible enough to support different interpolation algorithms 
- Use cases include cross-validation, hyperparameter tuning, and automated pipelines where input grids vary by sample

**Related sklearn concepts:**
- Metadata routing: https://scikit-learn.org/stable/modules/metadata_routing.html
- BaseEstimator with doc-link support for custom transformers
- TransformerMixin for consistent fit/transform interface
